libkrun/Makefile

```diff
@@ -1,8 +1,28 @@
 LIBKRUNTIME_CFLAGS =	-Wall -shared -fPIC
 COMMON_CFLAGS = -Wall -pedantic -std=gnu99
+SUDO = sudo
 
-ifeq ("${TRAVIS}", "true")
-	COMMON_CFLAGS += -DTRAVIS
+# Sadly this tool requires root
+VIRT_WHAT = ${SUDO} /usr/sbin/virt-what
+
+# Detect if we are virtualised (Linux only for now XXX)
+ifeq ($(shell uname -s),Linux)
+	VIRTUALISED = 0  # default off
+
+	VIRT_FACTS = $(shell ${VIRT_WHAT})
+	ifneq ("${VIRT_FACTS}","")
+		VIRTUALISED = 1
+	endif
+
+	ifeq ("${TRAVIS}","true")
```

Comment:
That's an early return. The remainder of the function checks `virt-what`:
https://github.com/softdevteam/krun/blob/f14b9ebf659d3f41b273d296ebe699fa3e75d304/krun/platform.py#L1204